digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/RoadAddressImporter.scala

```diff
@@ -0,0 +1,313 @@
+  }
+
+  private def fetchRoadAddressFromConversionTable(ely: Long, filter: String): Seq[ConversionRoadAddress] = {
+    conversionDatabase.withDynSession {
+      val tableName = importOptions.conversionTable
+      sql"""select tie, aosa, ajr, jatkuu, aet, let, alku, loppu, TO_CHAR(alkupvm, 'YYYY-MM-DD hh:mm:ss'), TO_CHAR(loppupvm, 'YYYY-MM-DD hh:mm:ss'),
+               TO_CHAR(muutospvm, 'YYYY-MM-DD hh:mm:ss'), ely, tietyyppi, linkid, kayttaja, alkux, alkuy, loppux,
+               loppuy, (linkid * 10000 + ajr * 1000 + aet) as id, ajorataid from #$tableName
+               WHERE ely=$ely AND aet >= 0 AND let >= 0 AND lakkautuspvm IS NULL #$filter """
+        .as[ConversionRoadAddress].list
+    }
+  }
+
+  //TODO this method is duplicated (or almost)
+  protected def fetchRoadAddressFromConversionTable(minLinkId: Long, maxLinkId: Long, filter: String): Seq[ConversionRoadAddress] = {
+    conversionDatabase.withDynSession {
+      val tableName = importOptions.conversionTable
+      sql"""select tie, aosa, ajr, jatkuu, aet, let, alku, loppu, TO_CHAR(alkupvm, 'YYYY-MM-DD hh:mm:ss'), TO_CHAR(loppupvm, 'YYYY-MM-DD hh:mm:ss'),
+               TO_CHAR(muutospvm, 'YYYY-MM-DD hh:mm:ss'), ely, tietyyppi, linkid, kayttaja, alkux, alkuy, loppux,
+               loppuy, (linkid * 10000 + ajr * 1000 + aet) as id, ajorataid from #$tableName
+               WHERE linkid > $minLinkId AND linkid <= $maxLinkId AND  aet >= 0 AND let >= 0 AND lakkautuspvm IS NULL #$filter """
+        .as[ConversionRoadAddress].list
+    }
+  }
+
+  private def generateChunks(linkIds: Seq[Long], chunkNumber: Long): Seq[(Long, Long)] = {
+    val (chunks, _) = linkIds.foldLeft((Seq[Long](0), 0)) {
+      case ((fchunks, index), linkId) =>
+        if (index > 0 && index % chunkNumber == 0) {
+          (fchunks ++ Seq(linkId), index + 1)
+        } else {
+          (fchunks, index + 1)
+        }
+    }
+    val result = if (chunks.last == linkIds.last) {
+      chunks
+    } else {
+      chunks ++ Seq(linkIds.last)
+    }
+
+    result.zip(result.tail)
+  }
+
+  protected def fetchChunkLinkIdsFromConversionTable(chunk: Int): Seq[(Long, Long)] = {
+    //TODO Try to do the group in the query
+    conversionDatabase.withDynSession {
+      val tableName = importOptions.conversionTable
+      val linkIds = sql"""select distinct linkid from #$tableName where linkid is not null order by linkid""".as[Long].list
+      generateChunks(linkIds, 25000l)
+    }
+  }
+
+  private val withOnlyCurrentRoadAddress: String = " AND loppupvm IS NULL "
+  private val withOnlyHistoryRoadAddress: String = " AND loppupvm IS NOT NULL "
+  private val withCurrentAndHistoryRoadAddressFromDate: String = " AND (loppupvm IS NULL OR (loppupvm IS NOT NULL AND TO_CHAR(loppupvm, 'YYYY-MM-DD') <= '$date')) "
+  private val withCurrentAndHistoryRoadAddress: String = ""
+
+  def importRoadAddress(ely: Long): Unit ={
```

Comment:
Is this already deprecated and replaced by the importRoadAddress()?

Reply:
If this can be removed, we can also remove:
116: private def fetchRoadAddressFromConversionTable(ely: Long, filter: String): Seq[ConversionRoadAddress]